diff --git a/src/plugins/plugin.js b/src/plugins/plugin.js
index 9fc2370150ecaf3809413efac6f040e03503637c..ec504cad9d837641785af9216be536e40f36e7e2 100644
--- a/src/plugins/plugin.js
+++ b/src/plugins/plugin.js
@@ -113,11 +113,11 @@ async function buildViteTargets(configFilePath, projectRoot, options, context) {
         var _viteConfig_build;
         targets[options.buildTargetName] = await buildTarget(options.buildTargetName, namedInputs, buildOutputs, projectRoot);
         // If running in library mode, then there is nothing to serve.
-        if (!((_viteConfig_build = viteConfig.build) == null ? void 0 : _viteConfig_build.lib)) {
+        //if (!((_viteConfig_build = viteConfig.build) == null ? void 0 : _viteConfig_build.lib)) {
             targets[options.serveTargetName] = serveTarget(projectRoot);
             targets[options.previewTargetName] = previewTarget(projectRoot);
             targets[options.serveStaticTargetName] = serveStaticTarget(options);
-        }
+        //}
     }
     // if file is vitest.config or vite.config has definition for test, create target for test
     if (configFilePath.includes('vitest.config') || hasTest) {
